src/initialization/initializeTypeScript/writeMultiTypeScriptConfig.ts

```diff
@@ -11,61 +12,74 @@ export interface MultiTypeScriptConfigSettings {
 	testFiles?: string;
 }
 
-export const writeMultiTypeScriptConfig = async ({
-	fileName,
+export const writeMultiTypeScriptConfig = async (
+	settings: MultiTypeScriptConfigSettings,
+) => {
+	const config = generateMultiTypeScriptConfig(settings);
+	await fs.writeFile(settings.fileName, JSON.stringify(config, undefined, 4));
+};
+
+export const generateMultiTypeScriptConfig = ({
 	improvements,
 	project,
 	sourceFiles,
 	testFiles,
 }: MultiTypeScriptConfigSettings) => {
-	await fs.writeFile(
-		fileName,
-		JSON.stringify(
-			[
-				{
-					fixes: {
-						...printImprovements(improvements),
-						strictNonNullAssertions: true,
-					},
-					...(testFiles && { include: [testFiles] }),
-					projectPath: project.filePath,
-					types: {
-						strictNullChecks: true,
-					},
-				},
-				{
-					...(testFiles && { exclude: [testFiles] }),
-					fixes: printImprovements(improvements),
-					...(sourceFiles && { include: [sourceFiles] }),
-					projectPath: project.filePath,
-				},
-				{
-					fixes: printImprovements(improvements),
-					...(testFiles
-						? { include: [testFiles, sourceFiles] }
-						: { include: [sourceFiles] }),
-					projectPath: project.filePath,
-				},
-			],
-			undefined,
-			4,
-		),
-	);
+	const fixes = printImprovements(improvements);
+
+	const stage1: Partial<RawTypeStatOptions> = {
+		fixes: {
+			...fixes,
+			strictNonNullAssertions: true,
+		},
+		projectPath: project.filePath,
+		types: {
+			strictNullChecks: true,
+		},
+	};
+
+	const stage2: Partial<RawTypeStatOptions> = {
+		fixes,
+		projectPath: project.filePath,
+	};
+
+	const stage3: Partial<RawTypeStatOptions> = {
+		fixes,
+		projectPath: project.filePath,
+	};
+
+	if (testFiles) {
+		// @ts-expect-error Cannot assign to 'include' because it is a read-only property.
+		stage1.include = [testFiles];
+		// @ts-expect-error Property 'exclude' does not exist on type 'Partial<PendingTypeStatOptions>'.
```

Comment:
[Bug] Err, what's going on with `exclude`? It's not used elsewhere - so I think it's another existing bug, right?